scarb/src/bin/scarb/commands/proc_macro_server.rs

```diff
@@ -1,7 +1,52 @@
 use anyhow::Result;
-use scarb::core::Config;
+use scarb::{
+    compiler::{plugin::proc_macro::ProcMacroHost, CairoCompilationUnit, CompilationUnit},
+    core::{Config, Workspace},
+    ops::{self, FeaturesOpts, FeaturesSelector},
+};
 
 #[tracing::instrument(skip_all, level = "info")]
-pub fn run(_config: &Config) -> Result<()> {
-    unimplemented!()
+pub fn run(config: &mut Config) -> Result<()> {
+    let ws = ops::read_workspace(config.manifest_path(), config)?;
```

Comment:
What if this changes in time (i.e. more crates are added)

Reply:
We will detect this on a client (LS) and just restart the server